proxy/grpcproxy/watch.go

```diff
@@ -104,6 +105,7 @@ func (sws *serverWatchStream) recvLoop() error {
 				ch: sws.watchCh,
 
 				progress: cr.ProgressNotify,
+				filters:  v3rpc.FiltersFromRequest(cr),
```

Comment:
can an unfiltered watcher on the same range be coalesced into a filtered watcher?